paddle/fluid/framework/ir/graph_pattern_detector.cc

```diff
@@ -1001,9 +1004,11 @@ PDNode *patterns::Transpose2Dequant::operator()() {
   auto *transpose2_op =
       pattern->NewNode(transpose2_op_repr())->assert_is_op("transpose2");
   auto dequant_in = pattern->NewNode(dequant_in_repr())
+                        ->assert_has_n_inputs(1)
                         ->assert_is_op_input("dequantize", "Input");
-  auto dequant_op =
-      pattern->NewNode(dequant_op_repr())->assert_is_op("dequantize");
+  auto dequant_op = pattern->NewNode(dequant_op_repr())
+                        ->assert_is_op("dequantize")
+                        ->AsIntermediate();
```

Comment:
I think AsIntermediate has no effect on ops, only vars. I meant that you add this to the dequant_in above if it's removed.